marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -162,6 +162,13 @@ function setArtifact(artifactType, artifactName, artifact) {
     } catch (e) {}
     if (fn.empty(existingArtifact) && artifactLibrary.defaultArtifact) {
         artifact = Object.assign({}, artifactLibrary.defaultArtifact(artifactName, artifact.targetEntityType), artifact);
+        if(artifact.sourceName || artifact.sourceType) {
```

Comment:
I don't think this is needed. To find out,  there should be a JUnit test for this story that creates an ingestion step with sourceName and sourceType populated, and then the test runs the ingestion step to ensure that each ingested document (only need to ingest one, really) ends up with headers/source set correctly. 

To make that test pass - I don't think you need anything here. Instead, you just need to confirm that when the createHeaders is called by the default ingestion step module, sourceName and sourceType are both present in the options object (which I think will happen by default, I don't think you'll need to do anything for that). And then createHeaders will do all the work.